sc-memory/sc-core/sc-store/sc-event/sc_event_queue.c

```diff
@@ -95,23 +97,34 @@ void _sc_event_emission_pool_worker(sc_pointer data, sc_pointer user_data)
   _sc_event_emission_pool_worker_data_destroy(work_data);
 }
 
-void sc_event_emission_manager_initialize(sc_event_emission_manager ** manager, sc_uint32 max_events_and_agents_threads)
+void sc_event_emission_manager_initialize(sc_event_emission_manager ** manager, sc_memory_params const * params)
 {
   *manager = sc_mem_new(sc_event_emission_manager, 1);
   sc_queue_init(&(*manager)->deletable_events);
 
-  max_events_and_agents_threads = sc_boundary(max_events_and_agents_threads, 1, g_get_num_processors());
+  (*manager)->limit_max_threads_by_max_physical_cores = params->limit_max_threads_by_max_physical_cores;
+  (*manager)->max_events_and_agents_threads =
+      (*manager)->limit_max_threads_by_max_physical_cores
+          ? sc_boundary(params->max_events_and_agents_threads, 1, g_get_num_processors())
+          : sc_max(1, params->max_events_and_agents_threads);
   {
-    sc_message("[sc-events] Configuration:");
-    sc_message("\tMax events and agents threads: %d", max_events_and_agents_threads);
+    sc_memory_info("Sc-event managers configuration:");
+    sc_message(
+        "\tLimit max threads by max physical cores: %s",
+        (*manager)->limit_max_threads_by_max_physical_cores ? "Yes" : "No");
```

Comment:
```suggestion
        (*manager)->limit_max_threads_by_max_physical_cores ? "true" : "false");
```
Compatibility with config